libvast/src/system/application.cpp

```diff
@@ -49,4 +36,28 @@ application::application() {
   detail::adjust_resource_consumption();
 }
 
+void render_error(const application& app, const caf::error& err,
+                  std::ostream& os) {
+  if (err)
+    // The user most likely killed the process via CTRL+C, print nothing.
+    return;
+  os << render(err);
+  if (err.category() == caf::atom("vast")) {
+    auto x = static_cast<vast::ec>(err.code());
+    switch (x) {
+      default:
+        break;
+      case ec::invalid_subcommand:
+      case ec::missing_subcommand:
+      case ec::unrecognized_option:
+        auto ctx = err.context();
+        auto name = ctx.get_as<std::string>(1);
```

Comment:
The function `get_as` is unsafe to call without prior type-check. I'd add an if statement checking for `ctx.match_element<string>(1)`.